src/services/balancer/subgraph/entities/pools/index.ts

```diff
@@ -79,9 +88,40 @@ export default class Pools {
     } catch {
       // eslint-disable-previous-line no-empty
     }
+
+    if (this.excludedAddresses == null) {
+      // fetch only once
+      this.excludedAddresses = await this.getExcludedAddresses();
+    }
+
     return this.serialize(pools, pastPools, period, prices, currency);
   }
 
+  public removeExcludedAddressesFromTotalLiquidity(
+    pool: Pool,
+    rawTotalLiquidity: string
+  ) {
+    let totalLiquidity = bnum(rawTotalLiquidity);
```

Comment:
@garethfuller since we are going to refactor this later, I've made the decision to simplify and assume this function is executed 

For simplicity - this function assumes it was ran after the decoration phase so that `excludedAddresses` is already available.